src/uwtools/config/formats/base.py

```diff
@@ -76,6 +79,26 @@ def _characterize_values(self, values: dict, parent: str) -> tuple[list, list]:
                 complete.append(f"{INDENT}{parent}{key}")
         return complete, template
 
+    @staticmethod
+    def _compare_config_get_lines(d: dict) -> list[str]:
+        """
+        Returns a line-by-line YAML representation of the given dict.
+
+        :param d: A dict object.
+        """
+        sio = StringIO()
+        yaml.safe_dump(d, stream=sio, default_flow_style=False, indent=2, width=inf)
+        return sio.getvalue().splitlines(keepends=True)
+
+    @staticmethod
+    def _compare_config_log_header() -> None:
+        """
+        Log a visual header and description of diff markers.
+        """
+        log.info("-" * MSGWIDTH)
+        log.info("? => info, -/+ => line only in - or + file, blank => matching line")
+        log.info("-" * MSGWIDTH)
```

Comment:
These two methods are just helpers for `compare_config()` -- code factored out of that method to keep it reasonably abstract.